extensions/modules/mail/src/main/java/org/exist/xquery/modules/mail/MessageListFunctions.java

```diff
@@ -476,68 +475,56 @@ private SearchTerm parseSearchTerms( Node terms ) throws XPathException
 	{
 		SearchTerm	st = null;
 		
-		if( terms.getNodeType() == Node.ELEMENT_NODE && "searchTerm".equalsIgnoreCase(terms.getLocalName()) ) {
+		if (terms.getNodeType() == Node.ELEMENT_NODE && "searchTerm".equalsIgnoreCase(terms.getLocalName()) ) {
 			String type  = ((Element)terms).getAttribute( "type" );
 			
-			if( type != null ) {
-				if( "not".equalsIgnoreCase(type) ) {
-					st = new NotTerm( parseChildSearchTerm( terms ) );
-				} else if( "and".equalsIgnoreCase(type) ) {
-					st = new AndTerm( parseChildSearchTerms( terms ) );
-				} else if( "or".equalsIgnoreCase(type) ) {
-					st = new OrTerm( parseChildSearchTerms( terms ) );
-				} else if( "from".equalsIgnoreCase(type) ) {
-					st = parseFromTerm( terms );
-				} else if( "subject".equalsIgnoreCase(type) ) {
-					st = parseSubjectTerm( terms );
-				} else if( "body".equalsIgnoreCase(type) ) {
-					st = parseBodyTerm( terms );
-				} else if( "to".equalsIgnoreCase(type) || "recipient".equalsIgnoreCase(type) ) {
-					st = parseRecipientTerm( terms );
-				} else if( "header".equalsIgnoreCase(type) ) {
-					st = parseHeaderTerm( terms );
-				} else if( "flag".equalsIgnoreCase(type) ) {
-					st = parseFlagTerm( terms );
-				} else if( "sent".equalsIgnoreCase(type) ) {
-					st = parseSentDateTerm( terms );
-				} else if( "received".equalsIgnoreCase(type) ) {
-					st = parseReceivedDateTerm( terms );
-				} else {
-					throw( new XPathException(this, "Invalid Search Term type specified: " + type ) );
-				}
-			} else {
+			if (type.isEmpty()) {
 				throw( new XPathException(this, "Invalid Search Term type specified: null" ) );
 			}
-		} 
+			if ("not".equalsIgnoreCase(type) ) {
+				st = new NotTerm( parseChildSearchTerm( terms ) );
+			} else if ("and".equalsIgnoreCase(type) ) {
+				st = new AndTerm( parseChildSearchTerms( terms ) );
+			} else if ("or".equalsIgnoreCase(type) ) {
+				st = new OrTerm( parseChildSearchTerms( terms ) );
+			} else if ("from".equalsIgnoreCase(type) ) {
+				st = parseFromTerm( terms );
+			} else if ("subject".equalsIgnoreCase(type) ) {
+				st = parseSubjectTerm( terms );
+			} else if ("body".equalsIgnoreCase(type) ) {
+				st = parseBodyTerm( terms );
+			} else if( "to".equalsIgnoreCase(type) || "recipient".equalsIgnoreCase(type) ) {
+				st = parseRecipientTerm( terms );
+			} else if( "header".equalsIgnoreCase(type) ) {
+				st = parseHeaderTerm( terms );
+			} else if( "flag".equalsIgnoreCase(type) ) {
+				st = parseFlagTerm( terms );
+			} else if( "sent".equalsIgnoreCase(type) ) {
+				st = parseSentDateTerm( terms );
+			} else if( "received".equalsIgnoreCase(type) ) {
+				st = parseReceivedDateTerm( terms );
+			} else {
+				throw new XPathException(this, "Invalid Search Term type specified: " + type );
+			}
+		}
 		
-		if( st == null ) {
-			throw( new XPathException(this, "Invalid Search Terms specified" ) );
+		if (st == null) {
+			throw new XPathException(this, "Invalid Search Terms specified" );
 		}
```

Comment:
Maybe turn into a an switch statement. Also the null check seems to be obsolete here as the `getAttribute` should not return `null`

Reply:
@reinhapa yes, there is still a lot of code that can be optimised. But given the amount of code that is changed in this PR I would propose to do it in a separate PR